src/storage-controller/src/lib.rs

```diff
@@ -3199,11 +3237,58 @@ where
                 // Re-pack all rows
                 let mut packer = row_buf.packer();
                 packer.extend(unpacked_row.into_iter());
-                (row_buf.clone(), -1)
+
+                let update = Update {
+                    row: row_buf.clone(),
+                    timestamp: expected_upper.clone(),
+                    diff: -1,
+                };
+
+                update
```

Comment:
Does not seem to be covered by tests according to [coverage/428](https://buildkite.com/materialize/coverage/builds/428).

Reply:
This one is peculiar! The code was there before, I only added to it. I think it's only reached when a) there is enough updates in a status collection, and b) we do a restart of `envd`.